lib/rules/no-extra-semi.js

```diff
@@ -18,6 +18,8 @@ const astUtils = require("../util/ast-utils");
 
 module.exports = {
     meta: {
+        type: "layout",
```

Comment:
I think this should be `suggestion` since it removes `EmptyStatement` nodes from the AST.

Reply:
Agreed. This really probably should have been called "no-empty-statements", but alas.